File: apps/android/app/src/main/java/com/signalasi/chat/MicrosoftTtsVoiceCatalog.kt
package com.signalasi.chat

internal object MicrosoftTtsVoiceCatalog {
    const val XIAOXIAO = "zh-CN-XiaoxiaoNeural"
    const val XIAOXIAO_DRAGON_HD_FLASH = "zh-CN-Xiaoxiao:DragonHDFlashLatestNeural"
    const val XIAOXIAO2_DRAGON_HD_FLASH = "zh-CN-Xiaoxiao2:DragonHDFlashLatestNeural"

    val voices: List<String> = listOf(
        XIAOXIAO,
        XIAOXIAO_DRAGON_HD_FLASH,
        XIAOXIAO2_DRAGON_HD_FLASH
    )

    fun canonical(value: String?): String {
        val candidate = value.orEmpty().trim()
        return voices.firstOrNull { it.equals(candidate, ignoreCase = true) } ?: XIAOXIAO
    }
}

File: apps/ios/GalaxySSI/MicrosoftTTSVoiceCatalog.swift
import Foundation

enum MicrosoftTTSVoiceCatalog {
  static let xiaoxiao = "zh-CN-XiaoxiaoNeural"
  static let xiaoxiaoDragonHDFlash = "zh-CN-Xiaoxiao:DragonHDFlashLatestNeural"
  static let xiaoxiao2DragonHDFlash = "zh-CN-Xiaoxiao2:DragonHDFlashLatestNeural"

  static let voices = [
    xiaoxiao,
    xiaoxiaoDragonHDFlash,
    xiaoxiao2DragonHDFlash,
  ]

  static func canonical(_ value: String?) -> String {
    let candidate = value?.trimmingCharacters(in: .whitespacesAndNewlines) ?? ""
    return voices.first { $0.caseInsensitiveCompare(candidate) == .orderedSame } ?? xiaoxiao
  }
}
